### DISCUSSION TITLE
Chronicle of Techmino

### DISCUSSION THREAD

**Discussion by Trebor-Huang** [26F-Studio/Techmino#360] [Document]
> Life is easy to chronicle, but bewildering to practice.  -- Edward Morgan Foster

--------

This is an attempt to record the development history of Techmino. Feel free to edit the post for improvements. If GitHub doesn't allow you to do that, you can also leave comments.

It is known that a large portion of the code even in the earliest history of this repo came from other games MrZ had made. We shall not pursue that here, and therefore the record will start at 2019-07-18, as shown by the first commit version.

| Date | Version | Changes |
|-|-|-|
| 2019-06-26 | | The putative birthday of Techmino. It's accuracy remains to be investigated.
| 2019-07-18 | 0.0.190718 | The actual date in git is 2020-02-04. The earliest versions were recovered from the game files distributed to other players. A lot was already available. The game consisted of four modes: Marathon, Zen, 40 Lines and most notably, Battle with the 9S bots. A decent set of BGM and SFX was present, including `blank.ogg`, the main theme of Techmino.
| 2019-07-19 | 0.0.190719 | Note that different players in the Battle mode had different piece sequences. This version added a Solo mode with identical sequences, and an "asymmetric" solo mode with different ones. A Blind mode with pieces invisible after lock and a 20G Death mode were also added. As in the last version, no statistics were shown after the game ends.
| 2019-07-21 | 0.0.190721 | A GM Roll mode was added.
| 2019-07-26 | 0.0.190726 | Animations were added as background and on scene changes. The help screen was no longer empty, and began with the sentence "I think you don't need 'help'." `SRS.lua` was renamed to `TRS.lua`, signifying the birth of the new rotation system.
| 2019-07-27 | 0.0.190727 | "No animation, real loading!" Minor changes and refactoring.
| Unknown | 0.1 | Joystick support. The version numbers no longer reflect the date.
| | 0.2 | 
| | 0.3 | Touchscreen setting added. But (Verification needed) it was not used until the Android release came out.
| | 0.4 |
| | 0.5 | The mode selection changed from buttons to mechanical-counter-style. This allowed for a flexible number of modes.
| | 0.6 | Multiplayer for up to four players were added. Eight slots of keyboard mappings were provided.
| | 0.6.5 |
| | 0.6.8 |
| | 0.7.1 | SiYuan HeiTi font replaced the original Calibri font, presumably to avoid potential license issues. A Puzzle mode that would be what is now called Frictionless Mode was added.
| | 0.7.2 | Custom game was added. The player could adjust a dozen parameters to create a mode that suits their needs. Also, TSD and PC-puzzle mode were added. The Puzzle mode was deleted.
| | ... | 
| | 0.7.35 |
| | 0.8.0 |

--------------

A list of events to be dated, preferably using bisection:
- [x] (Creation of) O-spin (Fantastic Global Update II, v0.8.20)
- [ ] Touchscreen support (before 0.7.35)
- [ ] Adaptation of CC bot
- [ ] Continuous gravity (before 0.7.35)
- [ ] Changes of the Modes system
- [ ] Zframework (There's git commit message for this, wait till I get there)
- [ ] Addition of various BGM's and voice packs (very early)
- [ ] Multiplayer
- [ ] 626 day
- [ ] Drought mode (somewhere 0.7.2 - 0.7.35)
- [ ] Languages (somewhere 0.7.2 - 0.7.35)
- [ ] Tips (Somewhere 0.7.2 - 0.7.35)
- [ ] Demo bot player (0.7.2 - 0.7.35)
- [ ] Current mode selection (before 0.9.0)

**Comment by C29H25N3O5** [comment]
**Note: this thread is not being maintained and is HEAVILY outdated.**

[a comment by an automated account is not shown]
